cmd/insights/validate_opa.go

```diff
@@ -10,36 +10,81 @@ import (
+			if err != nil {
+				logrus.Error(err)
+			}
 			os.Exit(1)
 		}
-		fmt.Println("OPA policy validated successfully.")
+		if regoFileName != "" {
+			_, err := opavalidation.Run(regoFileName, objectFileName, fwrego.InsightsInfo{InsightsContext: insightsInfoContext, Cluster: insightsInfoCluster}, objectNamespaceOverride)
+			if err != nil {
+				fmt.Printf("OPA policy failed validation: %v\n", err)
+				os.Exit(1)
+			}
+			fmt.Println("OPA policy validated successfully.")
+		}
+
+		if batchDir != "" {
+			_, failedPolicies, err := opavalidation.RunBatch(batchDir, fwrego.InsightsInfo{InsightsContext: insightsInfoContext, Cluster: insightsInfoCluster}, objectNamespaceOverride)
+			fmt.Println() // separate output from RunBatch
+			if err != nil {
+				fmt.Printf("OPA policies failed validation: %v\n", err)
+				fmt.Printf("Please check the above output for details about the %s\n", opavalidation.HumanizeStringsOutput(failedPolicies, "failure"))
+				os.Exit(1)
+			}
+			fmt.Println("OPA policies validated successfully.")
+		}
 	},
 }
 
-func init() {
-	validateCmd.AddCommand(opaCmd)
-	opaCmd.Flags().StringVarP(&regoFileName, "rego-file", "r", "policy.rego", "The rego file to validate.")
-	opaCmd.Flags().StringVarP(&objectFileName, "kube-object-file", "k", "", "A Kubernetes manifest to provide as input to the rego policy.")
-	opaCmd.Flags().StringVarP(&objectNamespaceOverride, "object-namespace", "N", "", "A Kubernetes namespace to override any defined in the object-file.")
-	opaCmd.Flags().StringVarP(&insightsInfoCluster, "insightsinfo-cluster", "l", "test", "A Kubernetes cluster name returned by the Insights-provided insightsinfo() rego function.")
-	opaCmd.Flags().StringVarP(&insightsInfoContext, "insightsinfo-context", "t", "Agent", "An Insights context returned by the Insights-provided insightsinfo() rego function. The context returned by Insights plugins is typically one of: CI/CD, Admission, or Agent.")
-	err := opaCmd.MarkFlagRequired("kube-object-file")
-	if err != nil {
-		panic(fmt.Errorf("unable to mark flag persistent: %v", err))
+// checkValidateOPAFlags verifies supplied flags for `validate opa` are valid.
+func checkValidateOPAFlags() bool {
+	var validFlags bool = true
+	if batchDir == "" && regoFileName == "" {
+		logrus.Errorln("Please specify one of the --rego-file or --batch-directory options to validate one or more OPA policies.")
+		return false
+	}
+	if batchDir != "" {
+		if regoFileName != "" {
+			logrus.Errorln("Please specify only one of the --batch-directory or --rego-file options.")
+			return false
+		}
+		if objectFileName != "" {
+			logrus.Errorln("The --kube-object-file option is only used with the --rego-file option, to validate a single OPA policy.")
+			return false
+		}
+	}
+	if regoFileName != "" && objectFileName == "" {
+		logrus.Errorln("Please also specify a Kube object file when validating a single OPA policy, using the --kube-object-file option.")
+		return false
 	}
+	return validFlags
```

Comment:
```suggestion
	return true
```
and get rid of validFlags since you are not using or updating it anywhere.